lms/djangoapps/discussion/rest_api/api.py

```diff
@@ -353,9 +353,10 @@ def _get_user_profile_dict(request, usernames):
 
         A dict with username as key and user profile details as value.
     """
-    request.GET = request.GET.copy()  # Make a mutable copy of the GET parameters.
-    request.GET['username'] = usernames
-    user_profile_details = AccountViewSet.as_view({'get': 'list'})(request).data
+    req = request._request              # pylint: disable=protected-access
```

Comment:
Why this change? It seems like the original here should work according to the docs: https://docs.djangoproject.com/en/1.11/ref/request-response/#querydict-objects

It looks like the DRF request object is a wrapper around the django `HttpRequest` object?  Does it make more sense to use the `query_params` attribute instead?
https://www.django-rest-framework.org/api-guide/requests/#query_params

Reply:
The error in the test was:
```
AssertionError: The `request` argument must be an instance of `django.http.HttpRequest`, not `rest_framework.request.Request`.
```
The (controversial) incompatible change was in 3.7.4: https://github.com/encode/django-rest-framework/pull/5618

I'll look into there's another way to address it.

Reply:
I think the underlying issue is that invoking a view in a utility function is janky. `AccountViewSet.as_view({'get': 'list'})(req)` essentially just [calls `get_account_settings` and returns the result](https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/user_api/accounts/views.py#L289), so why not do that here instead of calling the view? It would avoid needing to patch the query params.

Reply:
This rings enough alarm-bell memories that I'd recommend searching the edx-platform codebase for `_request`. We've had to hack around this issue before IIRC.

Reply:
Hmm, I don't see any `request._request`.  I've changed to not use the view, and go straight to the api function instead.